components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/request/SAMLLogoutRequestFactory.java

```diff
@@ -0,0 +1,75 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.samlsso.logout.request;
+
+import net.shibboleth.utilities.java.support.net.URISupport;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.FrameworkClientException;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityRequestFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityRequest;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.processor.SAMLLogoutRequestProcessor;
+import org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.HTTP_POST_PARAM_SAML2_AUTH_REQ;
+
+/**
+ * This class checks whether requests from the Identity servlet are SAML Requests and
+ * provides a builder to an instance of SAMLLogoutRequest.
+ */
+public class SAMLLogoutRequestFactory extends HttpIdentityRequestFactory {
+
+    private static final Log log = LogFactory.getLog(SAMLLogoutRequestProcessor.class);
+
+    public SAMLLogoutRequestFactory() {
+
+        super();
+    }
+
+    @Override
+    public String getName() {
+
+        return "SAMLLogoutRequestFactory";
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
+
+        return StringUtils.isNotBlank(request.getParameter(SSOConstants.HTTP_POST_PARAM_SAML2_AUTH_REQ));
+    }
+
+    @Override
+    public IdentityRequest.IdentityRequestBuilder create(HttpServletRequest request, HttpServletResponse response)
+            throws FrameworkClientException {
+
+        SAMLLogoutRequest.SAMLLogoutRequestBuilder builder = new SAMLLogoutRequest.
+                SAMLLogoutRequestBuilder(request, response);
+        super.create(builder, request, response);
+        builder.isPost(StringUtils.isBlank(URISupport.getRawQueryStringParameter(request.getQueryString(),
+                HTTP_POST_PARAM_SAML2_AUTH_REQ)));
+        if (log.isDebugEnabled()) {
+            log.debug("Query string : " + request.getQueryString());
```

Comment:
assign `request.getQueryString()` to a variable and resuse

Reply:
Fixed with PR https://github.com/wso2-extensions/identity-outbound-auth-samlsso/pull/133